### PR TITLE
Move Array's SZGenericArrayEnumerator to shared partition

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Array.cs [hunk 1]
```diff
@@ -2142,4 +2142,61 @@ public void Reset()
             }
         }
     }
+
+    // This is a normal generic Enumerator for SZ arrays. It doesn't have any of the "this" stuff
```
Comment (jkoritzinsky) [inline src/System.Private.CoreLib/shared/System/Array.cs]: Is this comment still relevant?

**Comment by jkotas** [comment]
The CoreLib mirror PR should go in first.

**Comment by jkotas** [comment]
Could you please resolve the conflict?

**Comment by marek-safar** [comment]
Done

File: src/System.Private.CoreLib/shared/System/Array.cs [hunk 2]
```diff
@@ -2149,4 +2149,58 @@ public void Reset()
         }
 #endif // !CORERT
     }
+
+    internal sealed class SZGenericArrayEnumerator<T> : IEnumerator<T>
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Array.cs]: It was nested class before in CoreCLR. I would keep it as nested class.

Also, I would keep it under the `!CORERT` ifdef above. It won't be usable in CoreRT just yet.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Array.cs]: > keep it as nested class

I guess you do not want to take the SZArrayHelper with this.

Should it rather be just a separate file?

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Array.cs]: yeah, moved all enumerators to a new file

File: src/System.Private.CoreLib/shared/System/Array.Enumerators.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
+using System.Collections;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Array.Enumerators.cs]: License header

**Pull request merged**